objects/src/notes/tree.rs

```diff
@@ -0,0 +1,138 @@
+use miden_crypto::{
+    hash::rpo::RpoDigest,
+    merkle::{LeafIndex, MerkleError, SimpleSmt},
+};
+
+use super::NoteMetadata;
+use crate::{BATCH_OUTPUT_NOTES_TREE_DEPTH, BLOCK_OUTPUT_NOTES_TREE_DEPTH, MAX_NOTES_PER_BATCH};
+
+/// Wrapper over [SimpleSmt] for notes tree
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+pub struct NotesTree<const DEPTH: u8>(SimpleSmt<DEPTH>);
+
+impl<const DEPTH: u8> NotesTree<DEPTH> {
```

Comment:
I am not sure we need this common type: there may be enough differences between the block and the batch note trees where each deserve its own wrapper.

For example, I'm not sure we need `insert()` on the block note tree as we never actually insert notes one by one into the block note tree (we build a block not tree from batch note trees).